build-tools/packages/version-tools/src/internalVersionScheme.ts

```diff
@@ -405,6 +412,25 @@ export function getVersionRange(
 	return range;
 }
 
+function bumpLegacyCompatVersion(
+	version: semver.SemVer | string,
+	bumpRange: number,
+): semver.SemVer {
+	const semVersion = semver.parse(version);
+	if (!semVersion) {
+		throw new Error("Invalid version string");
+	}
+	const minor = Math.floor(semVersion.minor / 10) * 10 + bumpRange;
```

Comment:
With bump range written this way it would be too easy to generate overlapping legacy compat ranges.
x.10.y w/ range 30 : >= x.10.y < x.40.0
x.25.y w/ range 30 : >= x.25.y < x.50.0 which mean we can no longer do a legacy break at x.40.

Rather than a bump range, the argument could be `minorMultiple` (name feels questionable).
Wherever you have "legacyCompat" string option, perhaps make it `{ type: "legacyCompat"; minorMultiple: number }` or something like that.

In some tooling (release tooling?), it would be good to check that prior version doesn't use a legacy compat range that is longer than the one we are generating. Might be tricky; so, in lieu of that, having appropriate documentation with this additional support is warranted.